src/pytorch_lightning/utilities/data.py

```diff
@@ -322,12 +322,54 @@ def _dataloader_init_kwargs_resolve_sampler(
     batch_sampler = getattr(dataloader, "batch_sampler")
     is_predicting = mode == RunningStage.PREDICTING
     # checking the batch sampler type is different than PyTorch default.
-    if batch_sampler is not None and (type(batch_sampler) is not BatchSampler or is_predicting):
-        batch_sampler = type(batch_sampler)(
-            sampler,
-            batch_size=batch_sampler.batch_size,
-            drop_last=(False if is_predicting else batch_sampler.drop_last),
-        )
+    batch_sampler_cls = type(batch_sampler)
+    if batch_sampler is not None and (batch_sampler_cls is not BatchSampler or is_predicting):
```

Comment:
Hey @otaj. I am confused here. Why do we need to re0-instantiate the BatchSampler of the users ?

This was done for data loaders because they block settings arguments once instantiated. But here, we can modify it in-place.

Reply:
Hi @tchaton, fair question and to be honest, my best answer is "it was done like this when I came to this code". However, it might be better for custom batch samplers since they might do some setup in `__init__` based on received Sampler.